scarb/src/compiler/plugin/proc_macro/compilation.rs

```diff
@@ -61,6 +67,34 @@ impl SharedLibraryProvider for Package {
             .path_unchecked()
             .join(lib_name))
     }
+
+    fn prebuilt_lib_path(&self) -> Result<Utf8PathBuf> {
+        let target_triple = target!();
+
+        let prebuilt_name = format!(
+            "{name}_v{version}_{target}{suffix}",
+            name = self.id.name,
+            version = self.id.version,
+            target = target_triple,
+            suffix = DLL_SUFFIX
+        );
+
+        let prebuilt_path = self
+            .root()
+            .join("target")
+            .join("scarb")
+            .join("cairo-plugin")
+            .join(prebuilt_name);
+
+        prebuilt_path
+            .exists()
+            .then_some(prebuilt_path)
+            .ok_or_else(|| anyhow!("prebuilt library not found"))
+    }
+
+    fn is_prebuilt(&self) -> bool {
+        self.prebuilt_lib_path().is_ok()
+    }
```

Comment:
Personally, I find this function more confusing than helpful, as all it does is check if path exist - it still doesn't mean that macro can be actually loaded, or that the compilation will not be run, but the name does not reflect that. Maybe we should just inline `prebuilt_path().is_some()`  instead? Sounds short enough to me. 🤔